src/cpp/rtps/builtin/discovery/database/DiscoverySharedInfo.hpp

```diff
@@ -102,7 +102,7 @@ class DiscoverySharedInfo
     virtual void to_json(
             nlohmann::json& j) const;
 
-private:
+protected:
```

Comment:
Why is this changed to protected?

Reply:
It made sense before when there was a mutex in this class, so children could use it. Now it is not needed.
However, as it is a class to be inherited and the common implementation along the project is to use protected, I think it is better to leave it this way